### ISSUE TITLE
Adding element to a dimension / keep dimension after reduce

### ISSUE THREAD

**Issue by lforesta** [Open-EO/openeo-processes#54]
When doing some band mathematics and creating a new band, one can use reduce() and specify the needed formula as in [this example](https://open-eo.github.io/openeo-api/processgraphs/). However reduce() will remove that dimension (spectral in this case).

What if I want to add this new band to the ones already existing in the cube, in this case changing the cardinality of the spectral dimension? Is there a way to do this with the existing processes?

@m-mohr suggested to merge the output of the reduce operation to the original cube, however the newly created cube will not have the dimension over which it was reduced. Following this idea, one could use reduce() and then use add_dimension to re-introduce the lost dimension, then use merge_cubes()?

Is there a better way?

**Comment by m-mohr** [comment]
Yes, after checking the process descriptions I would say it could be solved like this (pseudo code):
```
cube = load_collection(...) // loads a data cube with dimensions x, y, t and spectral for example
// Do filtering etc.
new_band = reduce(cube, callback, 'spectral');
new_band = add_dimension(new_band, 'spectral', 'band_name', 'bands')
cube = merge_cubes(cube, new_band)
// Export data cube etc.
```

Strictly speaking, you don't want to reduce the data / remove a dimension, so reduce doesn't feel like the right process here. So I'm thinking whether apply_dimension with dimension set to spectral could also be a solution. It's not quite covered by the description, but would also seem logical to me.

**Comment by lforesta** [comment]
with apply_dimension the cardinality must not change, so I think the option reduce/add_dimension/merge_cubes is the way to go currently
It seems counter-intuitive to having to reduce a dimension and then add it again and then merge, so maybe in the future we could consider a wrapper-process for these three operations, unless someone has a better approach for this.

**Comment by m-mohr** [comment]
> with apply_dimension the cardinality must not change

Yes, that's what I meant by "It's not quite covered by the description". So in the current version

> the option reduce/add_dimension/merge_cubes is the way to go currently

The question is, how should be be in future? What would you suggest? What process should cover such a use case? A changed apply_dimension? A new process? Name? Description? ...

**Comment by claxn** [comment]
This was discussed some time ago and I am not sure whether this makes sense or if I got something wrong, but wouldn't such a concatenation of the processes above be easier if we don't remove the dimension, unless explicitly told by the user? I don't see any drawback in keeping dimensions with `length(dim) == 1`.

**Comment by claxn** [comment]
@m-mohr : Exactly, thats what I meant by "explicitly told by the user". I am fine with the optional argument `drop_dimension=false`, which could make things easier for us.

**Comment by m-mohr** [comment]
First draft available in PR #89.

**Comment by m-mohr** [comment]
Telco: We'll combine drop_dimension with target_dimension.

Behavior of target_dimension:
* string specified:
    * reducer returns a single value: New dimension added with a single label, old dimension gets dropped.
    * reducer returns multiple values: New dimension added with a multiple labels, old dimension gets dropped.
* null specified:
    * Dimension gets dropped.
    * Not allowed if reducer returns multiple values.
